cannot use POWEROFF and ADC at once

Hello,

## Environment

- Development Kit:      none
- Kit version (for WroverKit/PicoKit/DevKitC): v3
- Module or chip used:  ESP32
- IDF version (run ``git describe --tags`` to find it): ca3be15241
- Build System:         zephyr west
- Compiler version (run ``xtensa-esp32-elf-gcc --version`` to find it):
    // 1.22.0-80-g6c4433a
- Operating System:     Linux
- Using an IDE?: No
- Power Supply:         USB

## Problem Description

during linkage of my project I noticed the following error message:
```
/opt/zephyr-sdk-0.16.3/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32_zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(rtc_io_hal.c.obj): in function `rtcio_hal_set_direction':
/home/mark/embedded_dev/modules/hal/espressif/zephyr/esp_shared/components/driver/rtc_io_hal.c:23: multiple definition of `rtcio_hal_set_direction'; zephyr/libzephyr.a(rtc_io_hal.c.obj):/home/mark/embedded_dev/modules/hal/espressif/components/hal/rtc_io_hal.c:23: first defined here
/opt/zephyr-sdk-0.16.3/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32_zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(rtc_io_hal.c.obj): in function `rtcio_hal_isolate':
/home/mark/embedded_dev/modules/hal/espressif/zephyr/esp_shared/components/driver/rtc_io_hal.c:61: multiple definition of `rtcio_hal_isolate'; zephyr/libzephyr.a(rtc_io_hal.c.obj):/home/mark/embedded_dev/modules/hal/espressif/components/hal/rtc_io_hal.c:61: first defined here
/opt/zephyr-sdk-0.16.3/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32_zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(rtc_io_hal.c.obj): in function `rtcio_hal_set_direction_in_sleep':
/home/mark/embedded_dev/modules/hal/espressif/zephyr/esp_shared/components/driver/rtc_io_hal.c:70: multiple definition of `rtcio_hal_set_direction_in_sleep'; zephyr/libzephyr.a(rtc_io_hal.c.obj):/home/mark/embedded_dev/modules/hal/espressif/components/hal/rtc_io_hal.c:70: first defined here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

It can be boiled down to a problem where two versions of `rtc_io_hal.c` get added if both `CONFIG_POWEROFF` and `CONFIG_ADC` are enabled, namely:
- `zephyr/esp_shared/components/driver/rtc_io_hal.c` (one year old)
- `components/hal/rtc_io_hal.c` (three years old)

see also here and in the following lines:
https://github.com/zephyrproject-rtos/hal_espressif/blob/zephyr/zephyr/esp32/CMakeLists.txt#L180

However, I think this can be easily fixed if one replaces [line 188](https://github.com/zephyrproject-rtos/hal_espressif/blob/zephyr/zephyr/esp32/CMakeLists.txt#L188) with the following content:
`      ../esp_shared/components/driver/rtc_io_hal.c`


### Steps to reproduce

1. copy the deep_sleep sample from the zephyr repo
2. modify the project and enable an adc and use it in the sourcecode
3. build the project
... you'll see a similar error as above

--- a/zephyr/esp32/CMakeLists.txt
+++ b/zephyr/esp32/CMakeLists.txt
@@ -185,7 +185,7 @@ if(CONFIG_SOC_SERIES_ESP32 OR CONFIG_SOC_SERIES_ESP32_NET)
       ../esp_shared/components/esp_hw_support/sleep_modes.c
       ../esp_shared/components/esp_pm/pm_impl.c
       ../../components/esp_hw_support/port/esp32/rtc_sleep.c
-      ../../components/hal/rtc_io_hal.c
+      ../esp_shared/components/driver/rtc_io_hal.c
     )
   endif()
 

